File: leetcode/quests/data-structures-and-algorithms/array-2/q2.ts
function smallerNumbersThanCurrent(nums: number[]): number[] {
  const freq = new Uint16Array(101);

  for (const n of nums) {
    freq[n]++;
  }

  let sum = 0;
  let i = 0;
  while (i < freq.length) {
    const count = freq[i];
    freq[i] = sum;
    sum += count;
    i++;
  }

  const result: number[] = [];

  for (const n of nums) {
    result.push(freq[n]);
  }

  return result;
}

File: leetcode/quests/data-structures-and-algorithms/array-2/q2.go
package main

func smallerNumbersThanCurrent(nums []int) []int {
	freq := make([]int, 101)
	for _, n := range nums {
		freq[n]++
	}

	sum := 0
	for i := 0; i < len(freq); i++ {
		count := freq[i]
		freq[i] = sum
		sum += count
	}

	result := make([]int, len(nums))
	for i, n := range nums {
		result[i] = freq[n]
	}

	return result
}